neo/Persistence/Snapshot.cs

```diff
@@ -57,98 +59,41 @@ public virtual void Dispose()
         {
         }
 
-        private ECPoint[] _validators = null;
-        public ECPoint[] GetValidators()
+        public IEnumerable<(ECPoint PublicKey, BigInteger Votes)> GetRegisteredValidators()
         {
-            if (_validators == null)
+            byte[] script;
+            using (ScriptBuilder sb = new ScriptBuilder())
             {
-                _validators = GetValidators(Enumerable.Empty<Transaction>()).ToArray();
+                sb.EmitAppCall(NativeContract.NEO.ScriptHash, "getRegisteredValidators");
+                script = sb.ToArray();
+            }
+            using (ApplicationEngine engine = ApplicationEngine.Run(script, this, testMode: true))
+            {
+                return ((VMArray)engine.ResultStack.Peek()).Select(p =>
+                {
+                    Struct @struct = (Struct)p;
+                    return (@struct.GetByteArray().AsSerializable<ECPoint>(), @struct.GetBigInteger());
+                });
             }
-            return _validators;
         }
 
-        public IEnumerable<ECPoint> GetValidators(IEnumerable<Transaction> others)
+        private ECPoint[] _validators = null;
+        public ECPoint[] GetValidators()
         {
-            Snapshot snapshot = Clone();
-            foreach (Transaction tx in others)
+            if (_validators == null)
```

Comment:
What happend if in the same block, two tx check Validators, and change it, if we cache this maybe we could get wrong results.